buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -607,18 +608,18 @@ static void getBytes(AbstractByteBuf buf, long addr, int index, OutputStream out
         }
     }
 
-    private static void getBytes(long inAddr, byte[] in, int inOffset, int inLen, OutputStream out, int outLen)
+    private static void getBytes(MemoryAddress inAddr, byte[] in, int inOffset, int inLen, OutputStream out, int outLen)
             throws IOException {
         do {
             int len = Math.min(inLen, outLen);
             PlatformDependent.copyMemory(inAddr, in, inOffset, len);
             out.write(in, inOffset, len);
             outLen -= len;
-            inAddr += len;
+            inAddr = inAddr.add(len);
```

Comment:
rather than calling a native method each time, just keep inAddr as the base address. And increment the offset each time. Will need to add another function into PlatformDependant.

Reply:
I think that's a good shout.

There might be a few places where there are optimisations like this possible, as when I was porting I took the approach of get it working first, optimise second and I haven't yet got to the optimise stage